ikarus/finiteelements/mixin.hh

```diff
@@ -244,6 +247,36 @@ public:
      ...);
   }
 
+private:
+  template <typename Sk, typename BC, typename MT>
+  auto invokeSubscribeTo(BC& bc) {
+    // For Clang-16: we need the this-> otherwise the code in the if clause will never be called. For Gcc-12.2: with the
+    // this-> it throws a compiler error in certain cases
+#if defined(__clang__)
+    if constexpr (requires { this->Sk::template subscribeToImpl<MT>(); }) {
+#else
+    if constexpr (requires { Sk::template subscribeToImpl<MT>(); }) {
+#endif
+      auto fTuple = Sk::template subscribeToImpl<MT>();
```

Comment:
I would suggest that we not request a tuple of functions from the skill but we just forward, this, bc and the MT to the skill and the skill itself takes care to what it wants to subscribe. Then it can extract the non-linearsovler type from BC.